src/firmware_new/boards/frankie_v1/io/network_logger.c

```diff
@@ -0,0 +1,54 @@
+#include "firmware_new/boards/frankie_v1/io/network_logger.h"
+
+#include "FreeRTOS.h"
+#include "cmsis_os.h"
+#include "firmware_new/boards/frankie_v1/io/proto_multicast_communication_profile.h"
+#include "main.h"
+#include "pb.h"
+#include "pb_common.h"
+#include "pb_decode.h"
+#include "pb_encode.h"
+#include "shared/proto/robot_log_msg.nanopb.h"
+#include "task.h"
+
+// the id of the queue that contains tbots log protos
+static osMessageQueueId_t log_message_queue_id_;
+
+void io_network_logger_task(void* communication_profile)
+{
+    // holds the msg that was just dequeued
+    TbotsProto_RobotLog dequeued_robot_log_proto;
+
+    ProtoMulticastCommunicationProfile_t* profile =
+        (ProtoMulticastCommunicationProfile_t*)communication_profile;
+
+    osStatus_t status;
+
+    for (;;)
```

Comment:
What is the purpose of this io_network class? And is it okay to leave this running in an infinite loop?

Reply:
the io_network logger is responsible for consuming the queue of robot log msgs and sending them over the network

the infinite loop is a fairly common pattern in FreeRTOS, you want this task to keep doing this as long as the robot is alive. The `osMessageQueueGet` function blocks until there is a msg so it doesn't actually use up _too_ much resources